docs/content/getting-started/production-best-practices.mdx

```diff
@@ -32,6 +33,16 @@ In addition, it's strongly recommended to fine-tune your server settings:
 
 - The server setting `OPENFGA_DATASTORE_MAX_OPEN_CONNS` should be set to be equal to your database's max connections. For example, in Postgres, you can see this value via running the SQL query `SHOW max_connections;`. If you are running multiple instances of the OpenFGA server, you should divide this setting equally among the instances. For example, if your database's `max_connections` is 100, and you have 2 OpenFGA instances, `OPENFGA_DATASTORE_MAX_OPEN_CONNS` should be set to 50 for each instance.
 
+## Concurrency Limits
+Check and ListObjects queries can potentially involve a high-degree of depth or breadth to resolve the query. If a query involves a lot of recursion or indirect group lookups, then it's possible that a single query could dominate the server's resources and the database connections. For this reason, we recommend setting some concurrency limits to protect OpenFGA servers.
```

Comment:
should we introduce the idea here that Check is like looking for a leaf node in a tree? otherwise, people won't understand what depth or breadth mean

Reply:
also, we mention breadth and depth but the settings we mention below don't put a limit on either of those 🤔

Reply:
I don't think it's this section of our documentation that should be responsible for describing how Check evaluation works. That should be saved for the concepts page or a blog post or some other form of documentation in our official docsite. The Production Best Practices should strive to remain succinct  and to the point.

> also, we mention breadth and depth but the settings we mention below don't put a limit on either of those 🤔
I don't understand what you mean by this? Could you elaborate?

Reply:
Reworked the documentation a lot. Let me know what you think.